File: SoftRhUsuariosBO/src/pe/edu/pucp/softrh/usuarios/bo/ClienteBO.java
package pe.edu.pucp.softrh.usuarios.bo;

import java.util.ArrayList;
import java.util.Date;
import pe.edu.pucp.softrh.usuarios.dao.ClienteDAO;
import pe.edu.pucp.softrh.usuarios.daoimp.ClienteDAOImp;
import pe.edu.pucp.softrh.usuarios.model.Cliente;
import pe.edu.pucp.softrh.usuarios.model.Usuario;

public class ClienteBO{
	private ClienteDAO clienteDAO;

	public ClienteBO() {
		this.clienteDAO = new ClienteDAOImp();
	}

	public Integer insertar(String dni, String nombres, String apellidos, String correo, String contrasenha, Date fechaRegistro, Boolean recibePromociones) {
		Usuario cliente = new Cliente(dni, nombres, apellidos, correo, contrasenha, fechaRegistro, recibePromociones);
		return clienteDAO.insertar((Cliente)cliente);
	}

	public Integer modificar(Cliente cliente) {
		return clienteDAO.modificar(cliente);
	}

	public Integer eliminar(Integer idCliente) {
		return clienteDAO.eliminar(idCliente);
	}

//	public ArrayList<Cliente> listarTodos() {
//		return clienteDAO.listarTodos();
//	}

	public Cliente obtenerPorId(Integer idCliente) {
		return clienteDAO.obtenerPorId(idCliente);
	}
}

File: Front/RHStoreWS/RHStoreUsuariosBO/ClienteBO.cs
using RHStoreBaseBO;
using RHStoreBaseBO.ServiciosWeb;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RHStoreUsuariosBO
{
	public class ClienteBO : BaseBO
	{
        public int insertar(string dni, string nombres, string apellidos, string correo, string contrasenha, DateTime fechaRegistro, bool recibePromociones)
        {
            return ClienteWS.insertarCliente(dni, nombres, apellidos, correo, contrasenha, fechaRegistro, recibePromociones);
        }

        public int modificar(int idUsuario, string dni, string nombres, string apellidos, string correo, string contrasenha, DateTime fechaRegistro, bool recibePromociones)
        {
            return ClienteWS.modificarCliente(idUsuario, dni, nombres, apellidos, correo, contrasenha, fechaRegistro, recibePromociones);
        }

        public int eliminar(int idCliente)
        {
            return ClienteWS.eliminarCliente(idCliente);
        }

        public BindingList<cliente> listarTodos()
        {
            return new BindingList<cliente>(ClienteWS.listarClientesTodos());
        }

        public cliente obtenerPorId(int idCliente)
        {
            return ClienteWS.obtenerClientePorId(idCliente);
        }
    }
}
